pkg/istio/istio.go

```diff
@@ -0,0 +1,144 @@
+package istio
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/deislabs/smi-metrics/pkg/mesh"
+
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/deislabs/smi-metrics/pkg/prometheus"
+	"github.com/deislabs/smi-sdk-go/pkg/apis/metrics"
+	"github.com/prometheus/client_golang/api"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+)
+
+type Config struct {
+	PrometheusURL    string  `yaml:"prometheusUrl"`
+	NamespaceQueries Queries `yaml:"namespaceQueries"`
+	PodQueries       Queries `yaml:"podQueries"`
+	WorkloadQueries  Queries `yaml:"workloadQueries"`
+}
+
+type Queries struct {
+	ResourceQueries map[string]string `yaml:"resourceQueries"`
+	EdgeQueries     map[string]string `yaml:"edgeQueries"`
+}
+
+type Istio struct {
+	NamespaceQueries Queries `yaml:"namespaceQueries"`
+	WorkloadQueries  Queries `yaml:"workloadQueries"`
+	PodQueries       Queries `yaml:"podQueries"`
+	prometheusClient promv1.API
+}
+
+func (l *Istio) GetSupportedResources(ctx context.Context) (*metav1.APIResourceList, error) {
```

Comment:
Good one but I thought as Linkerd and Istio may not answer all types of workload queries. 
I thought having separate ones would be greated. This would've been useful if `Pod` or some workload type queries were not possible.

Reply:
I think it makes a ton of sense to do it this way first, get it working and then look and see what's common. They're so similar it makes sense to move into a stdlib,  but it might have gone down differently.